### PR TITLE
Improvements to the dev deploy

### PR DESCRIPTION
**Pull request by kmcnellis**
### Description
- Fix the dev deploy to generate the terraform override in dev mode
- Allow the deploy script to be re-run to account for errors (and without re-creating the login radius cert)
- Add generated resources to the .gitignore
- Add output to track the script progress
- Add troubleshooting entries to the dev deploy readme
- Create a terraform deploy script to remove all remote resources
- Update secrets instructions
- Lint files

### Checklist
- [ ] Created tests which fail without the change (if possible)
- [x] Extended the README / documentation, if necessary

### CODE REVIEW

File: cloud/azure/templates/azure_saml_ses/main.tf [hunk 1]
```diff
@@ -10,7 +10,6 @@ terraform {
     }
     random = {}
   }
```
Comment (goldblatt) [inline cloud/azure/templates/azure_saml_ses/main.tf:12]: this needs to be included

File: cloud/deploys/dev_azure/README.md [hunk 2]
```diff
@@ -50,27 +50,76 @@ cloud/deploys/dev_azure/bin/setup --tag=<IMAGE_TAG>
...
 cloud/deploys/dev_azure/bin/deploy --tag=<IMAGE_TAG>
 ```
 or Within the app service resource, you can select Deployment Center, and within
 the registry settings change the 'Full Image Name and Tag' to be (the image_tag
 is what you specified in the build/tag/push)
 `<DOCKER_USERNAME>/<DOCKER_REPOSITORY>:<IMAGE_TAG>`
+
+## Troubleshooting
+
+### First step
```
Comment (goldblatt) [inline cloud/deploys/dev_azure/README.md:80]: this is mentioned above i think you can remove it here

File: cloud/azure/bin/setup-saml-keystore [hunk 3]
```diff
@@ -38,6 +39,14 @@ if [[ ! "${resource_group}" ]] \
   exit 1
 fi
 
+if  key_vault::has_secret "${vault_name}" "${KEYSTORE_PASS_SECRET_NAME}" && 
+    storage::has_blob "${storage_account_name}" \
+    "${KEYSTORE_CONTAINER_NAME}" \
+    "${KEYSTORE_FILENAME}" ; then
```
Comment (goldblatt) [inline cloud/azure/bin/setup-saml-keystore:45]: i think you need to remove the space here

**Review by goldblatt — CHANGES_REQUESTED** [review]

File: cloud/azure/bin/setup-saml-keystore (hunk 3 above)
Comment (kmcnellis) [reply cloud/azure/bin/setup-saml-keystore:45]: done

Comment (goldblatt) [reply cloud/azure/bin/setup-saml-keystore:45]: ooo sorry i still see it!

File: cloud/azure/templates/azure_saml_ses/main.tf (hunk 1 above)
Comment (goldblatt) [reply cloud/azure/templates/azure_saml_ses/main.tf:12]: and this?

**Review by goldblatt — CHANGES_REQUESTED** [review]
i think maybe the changes weren't pushed up!

**Comment by kmcnellis** [comment]
> i think maybe the changes weren't pushed up!

Ooops!  I don't know how that happened.  Github claims the push happened first, but 🤷 

image

File: cloud/azure/bin/setup-saml-keystore (hunk 3 above)
Comment (kmcnellis) [reply cloud/azure/bin/setup-saml-keystore:45]: OK, it's def gone now!

**Comment by kmcnellis** [comment]
Added a new destroy script (since you need a bastion ssh key to run the destroy command), and tweaked the secrets prompt a bit

File: cloud/azure/bin/setup-saml-keystore (hunk 3 above)
Comment (goldblatt) [reply cloud/azure/bin/setup-saml-keystore:45]: waitt is ittt?

Comment (goldblatt) [reply cloud/azure/bin/setup-saml-keystore:45]: "${KEYSTORE_FILENAME}" ; then space after quote and before semi-colon

**Review by goldblatt — CHANGES_REQUESTED** [review]
still see a few things off

**Review by goldblatt — COMMENTED** [review]
ooo i think you're github files got a lil confused here? seeing a lot of stuff from your other PR

**Comment by kmcnellis** [comment]
> ooo i think you're github files got a lil confused here? seeing a lot of stuff from your other PR

sigh.   I'm just going to start over.

**Comment by kmcnellis** [comment]
Split into the more-manageable https://github.com/seattle-uat/civiform/pull/2280 https://github.com/seattle-uat/civiform/pull/2279 and https://github.com/seattle-uat/civiform/pull/2278

**Pull request closed without merging**